### ISSUE TITLE
bundled libraries update

### ISSUE THREAD

**Issue by samth** [racket/racket#4497]
Here's an update on the current state of various bundled libraries. The full list of bundled libraries is here: https://github.com/racket/racket/blob/master/racket/src/native-libs/README.txt

I have not looked at the Linux natipkg-only libraries yet.

## Should upgrade
- [sqlite](https://www.sqlite.org/changes.html): now on 3400000. At least one [serious CVE](https://blog.trailofbits.com/2022/10/25/sqlite-vulnerability-july-2022-library-api/)
- 
## Maybe upgrade
- [openssl](https://www.openssl.org/): now on 1.1.1s and 3.0.7. Both the versions that we bundle have outstanding serious CVEs, but not in code that's used by Racket. The issue in 1.1.1o is less serious, and Windows AArch64 (the only place we use the 3.0 series) is a rare platform. 
- [zlib](https://www.zlib.net/): now on 1.2.13. CVE in code that we don't use
- [expat](https://github.com/libexpat/libexpat/blob/master/expat/Changes): now on 2.5.0. Multiple CVEs. Only used via pango and cairo.
- [pixman](https://gitlab.freedesktop.org/pixman/pixman/-/commits/master). Now on 0.42.2. Fixed a bunch of undefined behavior. 
- [poppler](https://gitlab.freedesktop.org/poppler/poppler/-/blob/master/NEWS): Now on 22.11.0. New numbering scheme. Our bundled version is 8 years old. One CVE, multiple other fuzzing-found security issues. Not clear where we actually use Poppler. 

## No need to upgrade now
- [libffi](https://github.com/libffi/libffi#history): now on 3.4.4; many bug fixes and feature improvements, no security issues
- [pkg-config](https://gitlab.freedesktop.org/pkg-config/pkg-config/-/blob/master/NEWS): now on 0.29.2; bug fixes but nothing significant
- [libedit](https://www.thrysoee.dk/editline/): updated but nothing interesting
- [libiconv](https://savannah.gnu.org/news/?group_id=424): no real change
- [gettext](https://git.savannah.gnu.org/gitweb/?p=gettext.git;a=blob;f=NEWS;hb=HEAD): now on 0.21.2. Maybe some build improvements.
- [glib](https://gitlab.gnome.org/GNOME/glib/-/blob/main/NEWS): Now on 2.75.0. Many changes; no mention of security.
- [libpng](http://www.libpng.org/pub/png/libpng.html). Now on 1.6.38. No security updates.
- [cairo](https://gitlab.freedesktop.org/cairo/cairo/-/blob/master/NEWS): Now on 1.17.6. No security updates. Soon to require Meson to build.
- [jpegsrc](https://www.ijg.org/files/): No new release.
- [harfbuzz](https://github.com/harfbuzz/harfbuzz/blob/main/NEWS): Now on 5.3.1 (we're quite behind). No security updates.
- [fribidi](https://github.com/fribidi/fribidi/blob/master/NEWS): Now on 1.0.12. Various fuzzing fixes and build fixes.
- [fontconfig](https://gitlab.freedesktop.org/fontconfig/fontconfig/-/blob/main/README). Now on 2.14.1. No security issues mentioned, but some UAF bugs fixed.
- [freetype](http://freetype.org/). No new releases.
- [pango](https://gitlab.gnome.org/GNOME/pango/-/blob/main/NEWS): Now on 1.50.12. Lots of changes. Nothing stands out. Still have to deal with Meson eventually. 
- [mpfr](https://gitlab.inria.fr/mpfr/mpfr/-/blob/master/NEWS): Now on 4.2.0. Lots of changes, seems to have some backwards-incompatible changes in memory handling. No security issues mentioned.
- [gmp](https://gmplib.org/gmp6.2). Still on 6.2.1.
- [atk](https://gitlab.gnome.org/GNOME/atk/-/blob/master/NEWS). Now on 2.38. No interesting changes. Still have to deal with Meson eventually.
- [MMTabBarView](https://github.com/MiMo42/MMTabBarView/commits/master). No releases. No security-relevant changes.
- [PSMTabBarControl](https://github.com/dorianj/PSMTabBarControl). Last changed in 2013.

**Comment by soegaard** [comment]
> [poppler](https://gitlab.freedesktop.org/poppler/poppler/-/blob/master/NEWS): Now on 22.11.0. New numbering scheme. Our bundled version is 8 years old. One CVE, multiple other fuzzing-found security issues. Not clear where we actually use Poppler.

Poppler is used to turn LaTeX formulas into picts (via pdf). 
I am using it it daily for my math web-site.

**Comment by soegaard** [comment]
Apropos bundled libraries: 

Thanks to Cairo`racket/draw` can output svg files.
There is however no way to turn an svg into a pict.
Asumu's `rsvg` which contains bindings for `librsvg` solves that problem,
since `librsvg` can render svgs onto Cairo contexts. 
It doesn't contain any binaries though.

Adding `librsvg` to the set of bundled libraries would ne nice.
The license is LGPLv2.1.

**Comment by samth** [comment]
@soegaard right, but that's in the popper package which is not part of main-distribution. I don't think there are any uses in the main distribution unless I've missed something.

**Comment by soegaard** [comment]
@samth It might not be used in the main distribution, but there is a reason it's listed in native-libraries.
Poppler depends on shared libraries in the main distribution (Cairo, Pango etc), so it is necessary 
to build it  at the same time these are updated.

**Comment by shhyou** [comment]
> Here's an update on the current state of various bundled libraries. The full list of bundled libraries is here: https://github.com/racket/racket/blob/master/racket/src/native-libs/README.txt
> I have not looked at the Linux natipkg-only libraries yet.
> ## No need to upgrade now
> * [MMTabBarView](https://github.com/MiMo42/MMTabBarView/commits/master). No releases. No security-relevant changes.

For the record, the last update to MMTabBarView by mflatt was on Apr 9, 2022 (racket/libs@8c1f814). I suppose the update includes bug fixes at least up to MiMo42/MMTabBarView#83.

**Comment by LiberalArtist** [comment]
> 
> - [jpegsrc](https://www.ijg.org/files/): No new release.
> 

I don't know many details, but AIUI most GNU/Linux distributions and browsers have switched from the IJG libjpeg to the libjpeg-turbo fork, which in now an ISO/ITU-T reference implementation: https://libjpeg-turbo.org It is API- and ABI-compatible with the IJG library, so it should be a drop-in replacement.

**Comment by shhyou** [comment]
Other related requests: racket#4395 and racket#4457
